stress/shell/src/main/java/alluxio/stress/cli/client/StressClientIOBench.java

```diff
@@ -162,11 +177,129 @@ public ClientIOTaskResult runLocal() throws Exception {
     service.awaitTermination(30, TimeUnit.SECONDS);
 
     ClientIOTaskResult.ThreadCountResult result = context.getResult();
+
     LOG.info(String.format("thread count: %d, errors: %d, IO throughput (MB/s): %f", numThreads,
         result.getErrors().size(), result.getIOMBps()));
+
     return result;
   }
 
+  /**
+   * Read the log file from java agent log file.
+   *
+   * @return summary statistics
+   * @throws IOException
+   */
+  @SuppressFBWarnings(value = "DMI_HARDCODED_ABSOLUTE_FILENAME")
+  public synchronized SummaryStatistics addAdditionalResult()
+      throws IOException, DataFormatException {
+    Map<String, PartialResultStatistic> methodDescToHistogram = new HashMap<>();
+
+    try (final BufferedReader reader = new BufferedReader(
+        new FileReader(BaseParameters.AGENT_OUTPUT_PATH))) {
+      String line;
+
+      final ObjectMapper objectMapper = new ObjectMapper();
+      while ((line = reader.readLine()) != null) {
+        final Map<String, Object> lineMap;
+        try {
+          lineMap = objectMapper.readValue(line, Map.class);
+        } catch (JsonParseException | MismatchedInputException e) {
+          // skip the last line of a not completed file
+          break;
+        }
+
+        final String type = (String) lineMap.get("type");
+        final String methodName = (String) lineMap.get("methodName");
+        final Integer duration = (Integer) lineMap.get("duration");
+
+        if (type == null || methodName == null || duration == null) {
+          continue;
+        }
+
+        if ((type.equals("AlluxioBlockInStream") && methodName.equals("readChunk"))
+            || (type.equals("HDFSPacketReceiver") && methodName.equals("doRead"))) {
+          if (!methodDescToHistogram.containsKey(methodName)) {
+            methodDescToHistogram.put(methodName, new PartialResultStatistic());
+          }
+
+          final PartialResultStatistic statistic = methodDescToHistogram.get(methodName);
+          statistic.mTimeToFirstByteNs.recordValue(duration);
+          statistic.mNumSuccess += 1;
+
+          if (duration > statistic.mMaxTimeToFirstByteNs[0]) {
```

Comment:
I think we should ignore his max response time array (and maybe add a TODO to represent it correctly), or implement it correctly. The original purpose of it was to collect the max response time over time of the test, so if the array had 20 entries, the 1st entry would be the max response time of the first 1/20th of the run, the 2nd entry is the max response time for the 2nd 1/20th of the run, and so on.

I don't think this is collecting the max this way.

Reply:
I changed to the right implementation of max time to first byte array.